core/runtime/src/main/java/io/quarkus/runtime/ApplicationConfig.java

```diff
@@ -0,0 +1,31 @@
+package io.quarkus.runtime;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
+public class ApplicationConfig {
+
+    /**
+     * The name of the application.
+     * If not set, defaults to the name of the project.
+     */
+    @ConfigItem(defaultValue = "${project.name:unset}")
+    public String name;
+
+    /**
+     * The version of the application.
+     * If not set, defaults to the version of the project
+     */
+    @ConfigItem(defaultValue = "${project.version:unset}")
+    public String version;
+
+    public static boolean isNameUnset(String name) {
+        return name.contains("unset");
```

Comment:
shouldn't it be `equals()` instead of contains? What if you have `unset` in your project name or even if the artifact is called `unset` (which might be a nice project name)?

I would make it `__UNSET__` or something similar.

Reply:
Yes you are right. This got left over from previous stuff I was checking. Let me fix it

Reply:
Done